src/main/java/com/uipath/uipathpackage/UiPathDeploy.java

```diff
@@ -106,37 +110,49 @@ public BuildStepMonitor getRequiredMonitorService() {
-        String orchestratorTenantFormatted = env.expand(orchestratorTenant.trim()).isEmpty() ? util.getValue(rb, "UiPath.DefaultTenant") : escapePowerShellString(env.expand(orchestratorTenant.trim()));
+
+        FilePath packagePathFormatted = new FilePath(new File(envVars.expand(packagePath.trim())));
+        if (packagePathFormatted.isDirectory()) {
+            packagePathFormatted.copyRecursiveTo(tempRemoteDir);
+        } else {
+            packagePathFormatted.getParent().copyRecursiveTo(packagePathFormatted.getName(), tempRemoteDir); // can use copyTo instead of copyRecursiveTo
+        }
+
+        String orchestratorTenantFormatted = envVars.expand(orchestratorTenant.trim()).isEmpty() ? util.getValue(rb, "UiPath.DefaultTenant") : escapePowerShellString(envVars.expand(orchestratorTenant.trim()));
         StandardUsernamePasswordCredentials cred = CredentialsProvider.findCredentialById(credentialsId, StandardUsernamePasswordCredentials.class, run, Collections.emptyList());
         if (cred == null || cred.getUsername().isEmpty() || cred.getPassword().getPlainText().isEmpty()) {
             throw new AbortException("Invalid credentials");
         }
         String username = escapePowerShellString(cred.getUsername()); //not using escapeJava as it will corrupt the username and password
         String password = escapePowerShellString(cred.getPassword().getPlainText());
-        listener.getLogger().println("Opening Powershell Session");
         try {
-            String importModuleCommands = util.importModuleCommands(listener, env);
-            String deployPackCommand = String.format("Deploy -orchestratorAddress %s -tenant %s -username %s -password %s -packagePath %s -authType UserPass", escapePowerShellString(orchestratorAddress), orchestratorTenantFormatted, username, password, packagePathFormatted);
-            util.execute(listener, importModuleCommands, deployPackCommand);
-            listener.getLogger().println("Exiting Powershell Session");
-        } catch (IOException | URISyntaxException e) {
+            String importModuleCommands = util.importModuleCommands(tempRemoteDir, listener, envVars);
+            String deployPackCommand = String.format("Deploy -orchestratorAddress %s -tenant %s -username %s -password %s -packagePath %s -authType UserPass", escapePowerShellString(orchestratorAddress), orchestratorTenantFormatted, username, password, escapePowerShellString(tempRemoteDir.getRemote()));
+            if (!new PowerShell(getCommand(importModuleCommands, deployPackCommand)).perform((AbstractBuild<?, ?>) run, launcher, listener)) {
+                throw new AbortException("Failed to execute powershell session");
+            }
+        } catch (URISyntaxException e) {
             e.printStackTrace(listener.getLogger());
             throw new AbortException(e.getMessage());
+        } finally {
+            tempRemoteDir.deleteRecursive();
```

Comment:
Since this try-catch-finally is scoped only for the execution of the Powershell command, any exceptions in the code before this try-catch, will not clean the remove directory.